subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/ProjectReportTask.java

```diff
@@ -56,17 +57,15 @@
 @DisableCachingByDefault(because = "Not worth caching")
 public abstract class ProjectReportTask extends AbstractProjectBasedReportTask<ProjectReportTask.ProjectReportModel> {
 
-    private final TextReportRenderer renderer = new TextReportRenderer();
+    public ProjectReportTask() {
+        getRenderer().convention(new TextReportRenderer()).finalizeValueOnRead();
```

Comment:
This would close the ability to set the value, wouldn't it?

And the reason I think we should finalize the renderers is because in `render` method the renderer is mutated:
* https://github.com/gradle/gradle/blob/de819bdc164f0fdf8cb500e497b120f8fa917893/subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/TaskReportTask.java#L181-L186

I added it everywhere for consistency.

Reply:
> This would close the ability to set the value, wouldn't it?

Yes it would. I thought you finalize it, because the original implementation had it as `final`.